README.md

```diff
@@ -76,18 +116,207 @@ load(
+You can include and use it similarly as well:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker"
+)
+```
+
+### Multiplatform CodeChechecker analysis: `codechecker_suite()`
+
+This rule is functionally equivalent to `codechecker_test` but allows for running on multiple platforms via the `platforms` parameter.
+You can include and use it similarly as well:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker_suite"
+)
+```
+
+### `codechecker_config()`
+
+Using the Bazel rule `codechecker_config()` you can parse a CodeChecker [configuration file](https://github.com/Ericsson/codechecker/blob/master/docs/config_file.md).
+
+First, include the rule in your BUILD file:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker_config"
+)
+```
+
+Create a CodeChecker configuration file e.g. `config.json` (see example [here](https://github.com/Szelethus/codechecker_bazel/blob/readme_update/test/config.json)) and parse it using `codechecker_config()`. 
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    config_file = ":config.json"
+)
+```
+
+Alternatively, you can assemble a CodeChecker configuration without a config file using the rule:
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    analyze = [
+        "--enable=bugprone-dangling-handle",
+        "--enable=bugprone-fold-init-type",
+        "--enable=misc-non-copyable-objects",
+        "--report-hash=context-free-v2",
+    ]
+)
+```
+
+You can now configure your `codechecker`, `codechecker_suite`, `codechecker_test` and `code_checker_test` targets using the above configuration:
 
 ```python
 codechecker_test(
     name = "your_codechecker_rule_name",
+    config = "your_codechecker_config",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+### Per-file CodeChecker analysis: `code_checker_test()`
+> [!IMPORTANT]
+> The rule is still in prototype status and is subject to changes without notice.
+> You are free to experiment and report issues however!
+
+Instead of a single CodeChecker call, the bazel rule `code_checker_test()` invokes
+[`CodeChecker analyze`](https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/user_guide.md#analyze)
+_for each_ translation unit in the targets to analyze. The rule is intended to be
+able to enable incremental analyses and dispatching  analysis jobs to remote build
+agents.
+
+To use `code_checker_test()` include it to your BUILD file:
+
+```python
+load(
+    "@bazel_codechecker//src:code_checker.bzl",
+    "code_checker_test",
+)
+```
+
+Create a `code_checker_test()` target by passing other targets you'd like CodeChecker to analyze:
+
+```python
+code_checker_test(
+    name = "your_code_checker_rule_name",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+```bash
+bazel build ://your_code_checker_rule_name
+```
```

Comment:
See, the description is identical. I guess no point to provide it.

Reply:
I have reduced this bit so that it references back to `codechecker_test` a bit more and doesn't have as much redundancy.